hw/top_earlgrey/BUILD

```diff
@@ -141,6 +165,7 @@ fpga_cw310(
     base_bitstream = "//hw/bitstream/hyperdebug:bitstream",
     exec_env = "fpga_hyper310_rom_with_fake_keys",
     manifest = "//sw/device/silicon_creator/rom_ext:manifest_standard",
+    openocd_adapter_config = "//third_party/openocd:jtag_cmsis_dap_adapter_cfg",
```

Comment:
`:fpga_cw340` likely needs this assignment too. Otherwise, I believe it should behave like the hyper310.

Reply:
I've never used the Hyperdebug CMSIS-DAP interface, but I believe it should work just like on the CW310.
Additionally, the CW340 has an embedded FTDI, and this is the config file.
https://github.com/lowRISC/opentitan/blob/master/util/openocd/board/cw340_ftdi.cfg
This is how you connect:
https://github.com/lowRISC/opentitan/blob/master/doc/getting_started/setup_fpga.md#connecting-openocd

Reply:
Ah, for CMSIS-DAP to work, the hyperdebug would need to be connected to the IOR pins via the ribbon cable. It depends on the configuration in CI.

Reply:
This rule isn't meant to run in CI - it's meant to be used by developers who want to start an OpenOCD server:

```
bazel run //third_party/openocd:jtag_server --//third_party/openocd:jtag_env=<some env>
```

You can then use `gdb` (or `telnet`) to go debug.  The rule executes a script that automates the necessary setup to get JTAG connected.

Adding the adapter config to the rules isn't too much of a big deal.  Hopefully that's all that will be needed to make this rule useful to CW340 users.

Reply:
> Adding the adapter config to the rules isn't too much of a big deal. Hopefully that's all that will be needed to make this rule useful to CW340 users.

Yeah, selecting the adapter config might work best for CW340. It's a funky one where the ribbon cable's position determines which adapter to select, and that's sort of orthogonal to the execution environment. I'm not sure it's in our interest to support every permutation in general, but for the JTAG server w/ CW340, maybe it's nicer to be able to select any config directly, without the execution environment proxy.